package/MDAnalysis/analysis/contacts.py

```diff
@@ -773,3 +536,325 @@ def plot_qavg(self, filename=None, **kwargs):
+        if out:
+            out[:] = (d <= self.radius)
+        else:
+            out = (d <= self.radius)            
+        return out
+
+    def qN(self, q, out=None):
+        """Calculate native contacts relative to reference state.
+
+        *q* is the matrix of contacts (e.g. :attr:`~ContactAnalysis1.q`).
+
+        If *out* is supplied as a pre-allocated array of the correct
+        shape then it is filled instead of allocating a new one in
+        order to increase performance.
+
+        This method is typically only used internally.
+        """
+        if out:
+            np.logical_and(q, self.mask, out)
+        else:
+            out = np.logical_and(q, self.mask)
+        contacts = out.sum()
+        return contacts, float(contacts) / self.mask.sum()
+
+    def plot(self, filename=None, **kwargs):
+        """Plot q(t).
+
+        .. function:: ContactAnalysis1.plot([filename, ...])
+
+        If *filename* is supplied then the figure is also written to file (the
+        suffix determines the file type, e.g. pdf, png, eps, ...). All other
+        keyword arguments are passed on to :func:`pylab.plot`.
+        """
+        if not self.timeseries :
+            raise ValueError("No timeseries data; do 'Contacts.run()' first.")
+        x, y, _ = zip(*self.timeseries)
+
+        import matplotlib.pyplot as plt
+        kwargs.setdefault('color', 'black')
+        kwargs.setdefault('linewidth', 2)       
+
+        fig = plt.figure()
+        ax = fig.add_subplot(111)
+        ax.plot(x, y, **kwargs)
+        ax.set_xlabel(r"frame number $t$")
+        ax.set_ylabel(r"contacts $q_1$")
+
+        if filename:
+            fig.savefig(filename)            
+        else:
+            fig.show()
+
+
+    def plot_qavg(self, filename=None, **kwargs):
+        """Plot :attr:`Contacts.qavg`, the matrix of average contacts.
+
+        .. function:: Contacts.plot_qavg([filename, ...])
+
+        If *filename* is supplied then the figure is also written to file (the
+        suffix determines the file type, e.g. pdf, png, eps, ...). All other
+        keyword arguments are passed on to :func:`pylab.imshow`.
+        """
+        if not self.contact_matrix :
+            raise ValueError("No timeseries data; do 'Contacts.run()' first.")
+        # collapse on the time-axis
+        data = np.array(self.contact_matrix)
+        data = data.mean(axis=0)
+
+        import matplotlib.pyplot as plt
+        import matplotlib.cm as cm
+
+        kwargs['origin'] = 'lower'
+        kwargs.setdefault('aspect', 'equal')
+        kwargs.setdefault('interpolation', 'nearest')
+        kwargs.setdefault('vmin', 0)
+        kwargs.setdefault('vmax', 1)
+        kwargs.setdefault('cmap', cm.hot)
+
+        fig = plt.figure()
+        ax = fig.add_subplot(111)
+        cax = ax.imshow(data, **kwargs)
+
+        cbar = fig.colorbar(cax)
+
+        if filename:
+            fig.savefig(filename)            
+        else: 
+            fig.show()
```

Comment:
I don't get this one? As in, what if variable `filename` has a weird file ending?